scripts/peer/peer.py

```diff
@@ -0,0 +1,273 @@
+    # This differs per celltype
+    # remove any rows with NA values
+    merged_expr_covs = merged_expr_covs.dropna()
+    expression = merged_expr_covs.drop(
+        ['PC1', 'PC2', 'PC3', 'PC4', 'sex', 'age'], axis=1
+    )
+    covariates = merged_expr_covs[
+        ['sampleid', 'PC1', 'PC2', 'PC3', 'PC4', 'sex', 'age']
+    ]
+    # make sure samples in covariates and expression data are equal
+    expression.sampleid.equals(covariates.sampleid)
+    # True
+    # remove sampleid from both covariates and expression dfs
+    expression.drop('sampleid', axis=1, inplace=True)
+    covariates.drop('sampleid', axis=1, inplace=True)
+    # finally, make sure age and sex in covariate df are int
+    covariates[['sex', 'age']] = covariates[['sex', 'age']].astype(int)
+
+    # return expression data and covariates
+    return covariates.to_csv(index=False), expression.to_csv(index=False)  # type: ignore
+
+
+def get_at_index(obj, idx):
+    """
+    get object index
+    """
+
+    return obj[idx]
+
+
+def run_peer_job(job: hb.batch.job.Job, expression_file, covariates_file):
+    """
+    Run peer analysis, except because peer is in Python2, we can't take
+    advantage of the python_job concept in Batch. Instead, we'll rely on
+    the two inputs `expression_file` and `covariates_file` to be TSV files.
+    """
+
+    job.image(PEER_DOCKER)
+
+    job.command(f'echo "expressions file:" && head {expression_file}')
+    job.command(f'echo "covariates file:" && head {covariates_file}')
+
+    # write python script to container
+    job.command(
+        """
+cat <<EOT >> run_peer.py
+
+import peer
+import numpy as np
+import pandas as pd
+
+def run_peer(expression_file, covariates_file, factors_output_path, weights_output_path, precision_output_path, residuals_output_path):
+    \"""
+    Get covariate data for each cell type
+    \"""
+
+    print 'Loading data'
+
+    # load in data
+    expr = pd.read_csv(expression_file, header=None, skiprows=1)
+    covs = pd.read_csv(covariates_file, header=None, skiprows=1)
+
+    print 'Loaded data'
+    print covs
+
+    # Set PEER paramaters as per the PEER website
+    model = peer.PEER()
+    model.setPhenoMean(expr)
+    model.setCovariates(covs)
+    model.setNk(10)
+    model.update()
+
+    # Calculate and save the PEER factors
+    factors = model.getX()
+    np.savetxt(factors_output_path, factors, delimiter=',')
+    # Calculate and save the weights for each factor
+    weights = model.getW()
+    np.savetxt(weights_output_path, weights, delimiter=',')
+    # Calculate and save the precision values
+    precision = model.getAlpha()
+    np.savetxt(precision_output_path, precision, delimiter=',')
+    # Calculate and save the residuals
+    residuals = model.getResiduals()
+    np.savetxt(residuals_output_path, residuals, delimiter=',')
+
+
+if __name__ == "__main__":
+    import sys
+    run_peer(sys.argv[1], sys.argv[2], sys.argv[3], sys.argv[4], sys.argv[5], sys.argv[6])
```

Comment:
Maybe could simplify this with:

```suggestion
    run_peer(*sys.argv[1:])
```